fs/react-next-base
# PWA Basics

# Summary

#### A brief description of the pull request.

#### Resolves: #issue_number

#### Screenshots in case of UI changes

# Test plan

#### List of steps to manually test introduced functionality:

- Install [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk?hl=en) for testing the app
- Pull `pwa-1` branch
- Install dependencies `yarn`
- Build the app `yarn build`
- Run the app `yarn start`
- Go to localhost
- Open dev tools with "Lighthouse" tab
- Generate the PWA report as on screen bellow:
![image](https://user-images.githubusercontent.com/51324247/108039524-3c379380-704d-11eb-9ebc-5c15aac164ef.png)
- You will see the report that app is installable:
![image](https://user-images.githubusercontent.com/51324247/108039708-83258900-704d-11eb-9df6-995bce0f30bb.png)
- Let's try to install the app as in animation bellow:
![ezgif com-video-to-gif (27)](https://user-images.githubusercontent.com/51324247/108040439-59209680-704e-11eb-8f46-6ca104871bee.gif)


#### What do need to pay more attention to, some implicit changes that influence the app:

- Optimization, some scripts and advanced service workers will be in the next PRs, stay tuned!

# Review notes

#### While reviewing pull-request (especially when it's your pull-request), please make sure that:

- you named PR close to issue name and linked that issue
- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes
